client/apps/dashboard/src/app/pages/sale/pages/single-page/sale-single-page.component.ts

```diff
@@ -0,0 +1,35 @@
+import {ChangeDetectionStrategy, Component} from '@angular/core';
+import {Validators} from '@angular/forms';
+import {FirestoreCollections} from '@jf/enums/firestore-collections.enum';
+import {LangSinglePageComponent} from '../../../../shared/components/lang-single-page/lang-single-page.component';
+import {URL_REGEX} from '../../../../shared/const/url-regex.const';
+
+@Component({
+  selector: 'jfsc-sale-single-page',
+  templateUrl: './sale-single-page.component.html',
+  styleUrls: ['./sale-single-page.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SaleSinglePageComponent extends LangSinglePageComponent {
+  collection = FirestoreCollections.Sales;
+
+  public buildForm(data: any) {
+    this.form = this.fb.group({
+      id: [
+        {value: data.id, disabled: this.currentState === this.viewState.Edit},
+        [Validators.required, Validators.pattern(URL_REGEX)]
+      ],
+      name: [data.name || '', Validators.required],
+      description: [data.description || ''],
+      saleValue: ['', Validators.required],
+      percentageValue: [data.percentageValue || ''],
+      fixedValue: [data.fixedValue || ''],
+      startingDate: [data.startingDate || ''],
+      endingDate: [data.endingDate || ''],
+      type: [data.type || ''],
```

Comment:
What is this?

Reply:
Type of sale, is it unlimited or limited

Reply:
Let's switch this to also just be a boolean `limited: boolean` we should only use enums when there are more than 2 options.

Reply:
why?
```suggestion
      type: [data.type || ''],
```
You think COUPON CODE can be available without times set ?

Reply:
The sales aren't actually coupons. Those would be the discounts. 

https://github.com/Jaspero/fireshop/issues/75

They are also heavily in development at the moment though. 
The sales module is intended more as a generally available sale. For example a holiday sale you would set up and then include a number of products. Anyone would be able to purchase the products at a discounted price. Discounts, on the other hand, are coupon codes you would send out and they could be applied to a particular order.